src/internal.c

```diff
@@ -3761,6 +3761,12 @@ int EccSharedSecret(WOLFSSL* ssl, ecc_key* priv_key, ecc_key* pub_key,
     else
 #endif
     {
+#ifdef OPENSSL_EXTRA
+        ret = wc_ecc_check_key(pub_key);
```

Comment:
I'm having trouble following the logic of only checking an eco public key in OpenSSL compatibility mode.  Please explain.

Reply:
OK. DH has a WOLFSSL_VALIDATE_FFC_IMPORT define internally to check the DH public key.
Maybe a WOLFSSL_VALIDATE_ECC_IMPORT could be used?

Reply:
The check is expensive.
I'll use WOLFSSL_VALIDATE_ECC_IMPORT and only turn it on for OPENSSL_ALL.

Reply:
OK. No check OPENSSL_EXTRA || WOLFSSL_VALIDATE_ECC_IMPORT for both calls.